product-mini/platforms/linux-sgx/enclave-sample/Enclave/Enclave.cpp

```diff
@@ -341,152 +439,223 @@ wasm_runtime_get_module_hash(wasm_module_t module)
 }
 #endif
 
-static void
-handle_cmd_instantiate_module(uint64 *args, uint32 argc)
+bool
+ecall_handle_cmd_instantiate_module(uint16_t enclave_module_idx,
+                                    uint32_t stack_size, uint32_t heap_size,
+                                    char *error_buf, uint32_t error_buf_size,
+                                    uint16_t *wasm_module_inst_idx)
 {
-    uint64 *args_org = args;
-    EnclaveModule *enclave_module = *(EnclaveModule **)args++;
-    uint32 stack_size = *(uint32 *)args++;
-    uint32 heap_size = *(uint32 *)args++;
-    char *error_buf = *(char **)args++;
-    uint32 error_buf_size = *(uint32 *)args++;
+    bool ret = false;
+    EnclaveModule *enclave_module = nullptr;
     wasm_module_inst_t module_inst;
 
-    bh_assert(argc == 5);
+    if ((wasm_module_inst_idx == nullptr)
+        or !(enclave_module =
+                 (EnclaveModule *)gEnclaveModuleMgr.get(enclave_module_idx))) {
+        ret = false;
+        goto exit;
+    }
 
     if (!(module_inst =
               wasm_runtime_instantiate(enclave_module->module, stack_size,
                                        heap_size, error_buf, error_buf_size))) {
-        *(void **)args_org = NULL;
-        return;
+        ret = false;
+        goto exit;
     }
 
-    *(wasm_module_inst_t *)args_org = module_inst;
+    if (!gWasmModuleInstMgr.add(module_inst, wasm_module_inst_idx)) {
+        ret = false;
+        goto exit;
```

Comment:
Should deinstantiate module_inst before goto exit:
`wasm_runtime_deinstantiate(module_inst)`

Reply:
Yes